src/server.js

```diff
@@ -36,49 +36,54 @@ function processCookie(req, res) {
   res.end(JSON.stringify(query));
 }
 
+function redirectLocation(urlPath) {
+  const newUrl = new url.URL('./', `http://ignored-host.com${urlPath}`);
```

Comment:
Is there any reason not just to use urlPath that I'm missing?

Reply:
We want to go from '/connect?token=token' to '/', and '/path/connect?token=token' to '/path/'.  The different cases are covered in the unit test :-)